### DISCUSSION TITLE
Wouldn't it be better if the audio source was Spotify itself?

### DISCUSSION THREAD

**Discussion by ghost** [KRTirtho/spotube#554] [Q&A] [answered]
What happens when the audio source is YouTube, a song that does not correspond to the track is played, for example, a live version, a musical video or even a mix, So ideally, if it's a Spotify client, it should play the same song that appears on Spotify. "Sorry, it's not possible", then this app is not a Spotify client but a simple app to play music from YouTube and nothing will make me change my mind.

**Comment by KRTirtho** [comment, marked as answer]
I wish I could use Spotify's audio source. But Spotify doesn't expose such API for free users or premium users

> "Sorry, it's not possible", then this app is not a Spotify client but a simple app to play music from YouTube and nothing will make me change my mind.

It's your personal opinion. We have nothing to do with it

**Reply by AlphaCraft9658** [reply]
> I wish I could use Spotify's audio source. But Spotify doesn't expose such API for free users or premium users
> 
> > "Sorry, it's not possible", then this app is not a Spotify client but a simple app to play music from YouTube and nothing will make me change my mind.
> 
> It's your personal opinion. We have nothing to do with it

From a different project I have found some useful information: Librespot is a library which alllows for direct Spotify audio source access with a premium account. So it could optionally be implemented for Spotify Premium users. The only thing is that usage of it might not be 100% safe.
